app/models/normalize_primo_common.rb

```diff
@@ -53,9 +53,13 @@ def sanitize_authors(authors)
   end
 
   def author_link(author)
-    [ENV['MIT_PRIMO_URL'], '/discovery/search?query=creator,exact,',
-     author, '&tab=', ENV['PRIMO_MAIN_VIEW_TAB'], '&search_scope=all&vid=',
-     ENV['PRIMO_VID']].join
+    [ENV.fetch('MIT_PRIMO_URL', nil), '/discovery/search?query=creator,exact,',
+     clean_author(author), '&tab=', ENV.fetch('PRIMO_MAIN_VIEW_TAB', nil), '&search_scope=all&vid=',
+     ENV.fetch('PRIMO_VID', nil)].join
+  end
+
+  def clean_author(author)
```

Comment:
I exctrated this out thinking the solution may have been more complex than just encoding. I left it as it's own method as `author_link` has quite a bit going on. I'm not sure if this makes things better or worse. I'm happy to collapse it or leave it as is.

Reply:
Great idea. I didn't even notice the confusing naming until you mentioned it. Thanks!